wallet/walletrpc.c

```diff
@@ -740,14 +740,63 @@ static struct command_result *json_signpsbt(struct command *cmd,
 	struct wally_psbt *psbt, *signed_psbt;
 	struct utxo **utxos;
 	u32 *input_nums;
+	const char *utxo_string;
 	u32 psbt_version;
+	bool *sign_all;
 
 	if (!param_check(cmd, buffer, params,
 			 p_req("psbt", param_psbt, &psbt),
 			 p_opt("signonly", param_input_numbers, &input_nums),
+			 p_opt("utxo_string", param_string, &utxo_string),
+			 p_opt_def("unsafe_sign_all", param_bool, &sign_all, false),
 			 NULL))
 		return command_param_failed();
 
+	if (!input_nums && !utxo_string && !sign_all)
+		return command_fail(cmd, LIGHTNINGD, "Must specify which inputs"
+				    " to sign with signonly or utxo_string");
+
+	if (input_nums && utxo_string)
+		return command_fail(cmd, LIGHTNINGD, "Must specify which inputs"
+				    " to sign with either signonly or"
+				    " utxo_string, not both.");
+
+	if (sign_all) {
+		input_nums = tal_free(input_nums);
+		utxo_string = tal_free(utxo_string);
+	}
+
+	if (utxo_string)
+		input_nums = tal_arr(tmpctx, u32, 0);
+
+	while (utxo_string && strlen(utxo_string)) {
+		struct bitcoin_outpoint usr_outpoint;
+
+		if (strlen(utxo_string) < 66)
```

Comment:
we already have a parser for a similar data structure, `param_txout` from wallet/reservation.c. 
this will expect you to pass in ["txid:vout", "txid:vout", "txid:vout"]

I'd recommend at the very least we copy the error messages to be the same/similar as from that function, if not reuse it.